operate/webapp/src/main/java/io/camunda/operate/webapp/security/permission/PermissionsService.java

```diff
@@ -0,0 +1,287 @@
+   *     decisions, or a list of decisionId
+   */
+  public ResourcesAllowed getDecisionsWithPermission(final IdentityPermission identityPermission) {
+    return getResourcesWithPermission(
+        AuthorizationResourceType.DECISION_DEFINITION, identityPermission);
+  }
+
+  /**
+   * getResourcesWithPermission
+   *
+   * @return resources for which the user has the given permission; the result matches either all
+   *     resources, or a list of resourceIds
+   */
+  public ResourcesAllowed getResourcesWithPermission(
+      final AuthorizationResourceType resourceType, final IdentityPermission identityPermission) {
+    if (!permissionsEnabled()) {
+      return ResourcesAllowed.all();
+    }
+    if (!isAuthenticated()) {
+      return ResourcesAllowed.withIds(Set.of());
+    }
+    if (identityPermission == null) {
+      throw new IllegalStateException("Identity permission cannot be null");
+    }
+    final PermissionType permissionType = getPermission(identityPermission);
+    final Authorization authorization = new Authorization(resourceType, permissionType);
+    final SecurityContext securityContext = getSecurityContext(authorization);
+    final List<String> ids = authorizationChecker.retrieveAuthorizedResourceKeys(securityContext);
+
+    return ResourcesAllowed.withIds(new LinkedHashSet<>(ids));
+  }
+
+  private boolean isAuthorized(
+      String resourceId, AuthorizationResourceType resourceType, PermissionType permissionType) {
+    final Authorization authorization = new Authorization(resourceType, permissionType);
+    final SecurityContext securityContext = getSecurityContext(authorization);
+    return authorizationChecker.isAuthorized(resourceId, securityContext);
+  }
+
+  private SecurityContext getSecurityContext(Authorization authorization) {
+    return new SecurityContext(getAuthentication(), authorization);
+  }
+
+  private io.camunda.security.auth.Authentication getAuthentication() {
+    final Long authenticatedUserKey = getAuthenticatedUserKey();
+    final List<String> authorizedTenants = tenantsService.tenantIds();
+    // groups and roles will come later
+    return new io.camunda.security.auth.Authentication.Builder()
+        .user(authenticatedUserKey)
+        .tenants(authorizedTenants)
+        .build();
+  }
+
+  private Long getAuthenticatedUserKey() {
+    final Authentication requestAuthentication =
+        SecurityContextHolder.getContext().getAuthentication();
+    if (requestAuthentication != null) {
+      final Object principal = requestAuthentication.getPrincipal();
+      if (principal instanceof final CamundaUser authenticatedPrincipal) {
+        return authenticatedPrincipal.getUserKey();
+      }
+    }
+    return null;
+  }
+
+  private boolean isAuthenticated() {
+    return (getAuthenticatedUserKey() != null);
+  }
+
+  public boolean permissionsEnabled() {
+    return operateProperties.getIdentity().isResourcePermissionsEnabled();
```

Comment:
we should use the new common properties cf https://github.com/camunda/camunda/blob/main/security/security-core/src/main/java/io/camunda/security/configuration/SecurityConfiguration.java
You can autowire a `SecurityConfiguration` bean